packages/dd-trace/test/opentelemetry/metrics.spec.js

```diff
@@ -0,0 +1,1035 @@
+        assert.strictEqual(histogram.histogram.dataPoints[0].sum, 30)
+      })
+
+      setupTracer({ OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE: 'DELTA' })
+      const meter = metrics.getMeter('app')
+      meter.createHistogram('latency').record(10)
+      meter.createHistogram('latency').record(20)
+
+      setTimeout(() => { validator(); done() }, 150)
+    })
+
+    it('histograms support CUMULATIVE temporality', (done) => {
+      const validator = mockOtlpExport((decoded) => {
+        const histogram = decoded.resourceMetrics[0].scopeMetrics[0].metrics[0]
+        assert.strictEqual(histogram.histogram.aggregationTemporality, 2)
+        assert.strictEqual(histogram.histogram.dataPoints[0].count, 3)
+        assert.strictEqual(histogram.histogram.dataPoints[0].sum, 60)
+      })
+
+      setupTracer({ OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE: 'CUMULATIVE' })
+      const meter = metrics.getMeter('app')
+      meter.createHistogram('latency').record(10)
+      meter.createHistogram('latency').record(20)
+      meter.createHistogram('latency').record(30)
+
+      setTimeout(() => { validator(); done() }, 150)
+    })
+  })
+
+  describe('Case Insensitivity', () => {
+    it('meter names are case-insensitive', () => {
+      setupTracer()
+      const meter1 = metrics.getMeter('MyApp')
+      const meter2 = metrics.getMeter('myapp')
+      assert.strictEqual(meter1, meter2)
+    })
+
+    it('metric names are case-insensitive', (done) => {
+      const validator = mockOtlpExport((decoded) => {
+        const counter = decoded.resourceMetrics[0].scopeMetrics[0].metrics[0]
+        assert.strictEqual(counter.name, 'mymetric')
+        assert.strictEqual(counter.sum.dataPoints[0].asInt, 6)
+      })
+
+      setupTracer()
+      const meter = metrics.getMeter('app')
+      const c1 = meter.createCounter('MyMetric')
+      const c2 = meter.createCounter('mymetric')
+      c1.add(1)
+      c2.add(2)
+      meter.createCounter('MYMETRIC').add(3)
+
+      setTimeout(() => { validator(); done() }, 150)
+    })
+
+    it('different instrument types with same name are distinct', (done) => {
+      const validator = mockOtlpExport((decoded) => {
+        const metrics = decoded.resourceMetrics[0].scopeMetrics[0].metrics
+        assert.strictEqual(metrics.length, 2)
+        const counter = metrics.find(m => m.sum)
+        const histogram = metrics.find(m => m.histogram)
+        assert(counter, 'Should have counter')
+        assert(histogram, 'Should have histogram')
+        assert.strictEqual(counter.name, 'test')
+        assert.strictEqual(histogram.name, 'test')
+        assert.strictEqual(counter.sum.dataPoints.length, 1, 'Counter should have 1 data point')
+        assert.strictEqual(histogram.histogram.dataPoints.length, 1, 'Histogram should have 1 data point')
+        assert.strictEqual(counter.sum.dataPoints[0].asInt, 5)
+        assert.strictEqual(histogram.histogram.dataPoints[0].sum, 100)
```

Comment:
I know we have lots of code like that, while I would like to start prevent calling so many asserts on a single object. We should just use the partial helper (assertObjectContains, if I am not mistaken)